### PR TITLE
chore: improve archive WAL check

### LINKED ISSUE #818: improve the logic to check if an object storage already contains a WAL
Right now our logic for checking if any existing WAL exists in the storage is very error prone because we rely on the WAL numeration, we should improve it by:

- We stop looking at the WAL file name to trigger the check for an empty archive

- At the end of the init job add a flag file in the data directory. I.e., $PGDATA/.cnp-wal-never-archived

- If an instance manager sees the ConditionContinuousArchiving true, remove the flag file if exists

- At the start of the walarchive command, if the flag file exists, run the check for an empty archive

- If the walarchive command succeeds remove the flag file if exists

improvement proposal originally made by @mnencia  and @gbartolini

### PR DESCRIPTION
**Pull request by armru**
Closes #818

Signed-off-by: Armando Ruocco <armando.ruocco@enterprisedb.com>

### CODE REVIEW

File: pkg/management/postgres/restore.go [hunk 1]
```diff
@@ -654,7 +654,7 @@ func (info *InitInfo) checkBackupDestination(
 
 	// Instantiate the WALArchiver to get the proper configuration
 	var walArchiver *archiver.WALArchiver
-	walArchiver, err = archiver.New(ctx, cluster, env, walarchive.SpoolDirectory)
+	walArchiver, err = archiver.New(ctx, cluster, env, walarchive.SpoolDirectory, os.Getenv("PGDATA"))
```
Comment (mnencia) [inline pkg/management/postgres/restore.go:657]: Pgdata is already in the info struct

File: pkg/management/barman/archiver/archiver.go [hunk 2]
```diff
@@ -23,17 +23,24 @@ import (
...
 	"github.com/cloudnative-pg/cloudnative-pg/pkg/management/barman"
 	barmanCapabilities "github.com/cloudnative-pg/cloudnative-pg/pkg/management/barman/capabilities"
 	"github.com/cloudnative-pg/cloudnative-pg/pkg/management/barman/spool"
 	"github.com/cloudnative-pg/cloudnative-pg/pkg/management/execlog"
 	"github.com/cloudnative-pg/cloudnative-pg/pkg/management/log"
 )
 
+const (
+	// CheckWalArchiveFile the flag file that activates the wal archiver checks
+	CheckWalArchiveFile = ".check-wal-archive"
```
Comment (gbartolini) [inline pkg/management/barman/archiver/archiver.go:41]:
```suggestion
	CheckWalArchiveFile = ".check-empty-wal-archive"
```

Comment (armru) [reply pkg/management/barman/archiver/archiver.go]: done

**Review by sxd — APPROVED** [review]

File: pkg/management/postgres/restore.go [hunk 3]
```diff
@@ -209,6 +210,14 @@ func (info InitInfo) restoreDataDir(backup *apiv1.Backup, env []string) error {
 		return err
 	}
 	log.Info("Restore completed")
+
+	filePath := filepath.Join(info.PgData, archiver.CheckEmptyWalArchiveFile)
+	// We create the check empty wal archive file to tell that we should check if the
+	// destination path it is empty
+	if err := fileutils.CreateEmptyFile(filepath.Clean(filePath)); err != nil {
```
Comment (armru) [inline pkg/management/postgres/restore.go:217]: I think ConfigureNewInstance is invoked also inside the restore process, so maybe this is not needed here

Comment (sxd) [reply pkg/management/postgres/restore.go]: fixed

**Review by gbartolini — APPROVED** [review]

**Review by mnencia — APPROVED** [review]

**Pull request merged**